cli/operator/node.go

```diff
@@ -344,46 +358,80 @@ var StartNodeCmd = &cobra.Command{
 		operatorNode = operator.New(logger, cfg.SSVOptions, slotTickerProvider, storageMap)
 
 		if cfg.MetricsAPIPort > 0 {
-			go startMetricsHandler(cmd.Context(), logger, db, metricsReporter, cfg.MetricsAPIPort, cfg.EnableProfile)
+			go startMetricsHandler(ctx, logger, db, metricsReporter, cfg.MetricsAPIPort, cfg.EnableProfile)
 		}
 
 		nodeProber := nodeprobe.NewProber(
 			logger,
 			func() {
-				logger.Fatal("ethereum node(s) are either out of sync or down. Ensure the nodes are healthy to resume.")
+				logger.Fatal("Ethereum nodes are either out of sync or down. Make sure the nodes are healthy to resume.")
 			},
 			map[string]nodeprobe.Node{
-				"execution client": executionClient,
+				nodeprobe.ExecutionClientNode: executionClient,
 
 				// Underlying options.Beacon's value implements nodeprobe.StatusChecker.
 				// However, as it uses spec's specssv.BeaconNode interface, avoiding type assertion requires modifications in spec.
 				// If options.Beacon doesn't implement nodeprobe.StatusChecker due to a mistake, this would panic early.
-				"consensus client": consensusClient,
+				nodeprobe.ConsensusClientNode: consensusClient,
 			},
 		)
 
-		nodeProber.Start(cmd.Context())
-		nodeProber.Wait()
-		logger.Info("ethereum node(s) are healthy")
-
-		metricsReporter.SSVNodeHealthy()
-
-		eventSyncer := setupEventHandling(
-			cmd.Context(),
-			logger,
-			executionClient,
+		eventFilterer, err := executionClient.Filterer()
```

Comment:
Can we try to extract some lines to a function/method/entity to avoid increasing the size of this function? It's quite large and IMO we need to gradually decrease its size

Reply:
I agree that we want function size to be smaller (not larger) pretty much all the time, but this is not the only/best thing to optimize for - the more important thing is that general code structure stays predictable (easy to understand and navigate),

so in that sense grouping related code together rather than splitting it apart (possibly mixing it with unrelated things - aka mixing different abstraction layers) could be a better choice - and [above is a good example](https://github.com/ssvlabs/ssv/pull/1819#discussion_r1827037473) of why we might want to "untangle" code sometimes

and now that it is untangled - we can find a better way to refactor `StartNodeCmd.Run` - which I'm trying to do in https://github.com/ssvlabs/ssv/pull/1843 (better do it in a separate PR cause this one is getting larger and larger already, plus it's also better to do after we merge https://github.com/ssvlabs/ssv/pull/1820 cause it removes some code from there too)